src/testdir/test_functions.vim

```diff
@@ -990,6 +990,12 @@ func Test_Executable()
   elseif has('unix')
     call assert_equal(1, executable('cat'))
     call assert_equal(0, executable('nodogshere'))
+
+    new
+    exe 'lcd /'
+    call assert_equal(1, executable('usr/bin/cat'))
+    call assert_equal('/usr/bin/cat', exepath('usr/bin/cat'))
+    bwipe
```

Comment:
Could have a test for Windows, but I was not sure how to do this best.
Could use `let e = exepath("notepad")` and then `lcd` to `e[:2]` maybe?